test/revenue-analysis.test.ts

```diff
@@ -0,0 +1,19 @@
+import { CommandFactory } from "nest-commander"
+import { CliModule } from '../src/cli.module'
+import { Logger } from '../src/logger'
+import { AnalyzeRevenuesCommand } from "../src/commands/analyze-revenue.cmd"
+import path from "path"
+
+describe('revenue analysis', () => {
+  describe('running the whole flow', () => {
+    it('evaluates revenues correctly', async () => {
```

Comment:
(non-blocking)
I'm pretty sure that testing all of this is far from straightforward and we can't spend too much time on it (hence the non-blocking note), but the huge fixtures (and snapshot) are quite difficult to grasp and work with if someone else comes here to update or add tests.

If it's too complex/lengthy to construct small test data which could be verified, I think it might help to add some comments about what exactly to look for in the snapshot or something like that, so that other devs wouldn't see "evaluates revenues _correctly_" only and get stuck if the huge snapshot doesn't match anymore (without having some hints about what exactly this tests).

Reply:
I tried now for some time to reduce the dataset but I would have to design the auction inputs and results from scratch and it can be done but started to get time-consuming. I added at least a short comment.

One good thing is that you can just take real life data and update all fixtures.